lib/services/internalEngine/index.js

```diff
@@ -571,6 +570,41 @@ class InternalEngine {
     })
       .catch(error => Bluebird.reject(this.esWrapper.formatESError(error)));
   }
+
+  /**
+   * Apply default mapping to the collection but preserve existing kuzzle metadata
+   *
+   * @param {string} index
+   * @param {string} collection
+   * @returns {Promise.<object>} returns the updated default mapping
+   */
+  applyDefaultMapping (index, collection, mapping) {
+    const defaultMapping = mapping;
+
+    return this.getMapping({ index, type: collection }, true)
+      .then(mappings => {
+        const existingMapping = mappings[index].mappings[collection];
+
+        // Preserve old version of kuzzle metadata mapping
+        if (existingMapping.properties && existingMapping.properties._kuzzle_info) {
```

Comment:
Isn't this a duplicate behavior of https://github.com/kuzzleio/kuzzle/pull/1257/files#diff-8927ec75654a199501c9220ef47f3b5cR1255 ?

Reply:
Yes but only this part to preserve existing _kuzzle_info

Reply:
I don't understand

Reply:
The only duplicated part is 
```
        // Preserve old version of kuzzle metadata mapping
        if (existingMapping.properties && existingMapping.properties._kuzzle_info) {
          Object.assign(
            defaultMapping._kuzzle_info.properties,
            existingMapping.properties._kuzzle_info.properties
          );
}
```
Apart from this part, the function are entirely different (one is just merging mapping and the other get an existing mapping from DB and then apply Kuzzle default to the collection mapping

Reply:
Dude, I think we should do some peer review of this because I still don't get the purpose of these two functions.